xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -1159,9 +1161,22 @@ bool CPVRClients::UpdateAndInitialiseClients(bool bInitialiseAllClients /* = fal
     }
   }
 
+  // indicate that addons have now been initialised
+  {
+    CSingleLock lock(m_critSection);
+    m_bAddonsInitialised = true;
```

Comment:
Ah, good catch, the wait() is supposed to use it. I guess it works without it too, but it's not very obvious.